### ISSUE TITLE
Syntax errors break the web Projection Boxes

### ISSUE THREAD

**Issue by KasraF** [UCSD-PL/vscode#34]
This code:
```py
import matplotlib.pyplot as plt
import numpy as np
from typing import List

def plot(data: List[int]):
    """
    Plot the given list of integers as a bar chart.
    Use flower names for the x-axis
    """
    ## ---
    plt.bar(np.arange(len(data)), data)
    plt.xticks(np.arange(len(data)), ['rose', 'tulip', 'daisy', 'orchid']
    ## ---


plot([1, 86, 42, 54, 100])
```

Correctly shows an exception on the local build:
![image](https://user-images.githubusercontent.com/11238981/222542061-07078049-bae0-4e9f-b13f-e14484bd728d.png)

But on the web, they fail silently:
![image](https://user-images.githubusercontent.com/11238981/222542178-db09a71a-8789-4d89-856d-c7af8070429d.png)

**Comment by KasraF** [comment]
Even locally, there's a corner case where if this code is the first to load, the Projection Boxes simply never show up, which might confuse the user. But at least even then they can see the exception in the `Run` panel. 

It might be worth handling syntax errors in a more robust way such that they appear _in the projection boxes_ similar to runtime errors both locally and on the web.

**Comment by KasraF** [comment]
Turns out the `remove_comments_and_docstrings` function was throwing an unhandled parse error, which _happened_ to work in the local version. We need to be more careful with error handling, there are a lot of unwritten assumptions about what can fail in `run.py` and when.

Fixed in efd0bce.
